wapitiCore/attack/mod_brute_login_form.py

```diff
@@ -0,0 +1,191 @@
+from itertools import chain
+from os.path import join as path_join
+import requests
+from bs4 import BeautifulSoup
+from requests.exceptions import ReadTimeout
+from wapitiCore.net.web import Request
+from wapitiCore.attack.attack import Attack
+from wapitiCore.language.vulnerability import Vulnerability, Anomaly, _
+from wapitiCore.net import web
+
+
+class mod_brute_login_form(Attack):
+    time_to_sleep = ''
+    name = "brute_login_forms"
+    payloads_username = []
+    payloads_password = []
+    PAYLOADS_FILE = "passwords.txt"
+    PAYLOADS_FILE_USER = "users.txt"
+    PAYLOADS_SUCCESS = "successMessage.txt"
+    PAYLOADS_FAIL = "incorrectMessage.txt"
+    MSG_VULN = _("Brute force attack success")
+    current_request_url = None
+    current_request_method = None
+    password_parameter = []
+    username_parameter_field = []
+    submit_var_name = None
+    submit_var_value = None
+
+    def set_timeout(self, timeout):
+        self.time_to_sleep = str(1 + int(timeout))
+
+    def has_password_field(self):
+
+        list = ['email','user','mail','session','login']
+        url = self.persister.get_root_url()
+        request = Request(url)
+        response = self.crawler.get(request, follow_redirects=True)
+        content = response.content
+        inputs = BeautifulSoup(content, features="lxml").findAll('input')
+        if self.current_request_method == 'POST':
+            for _input in inputs:
+
+                if (_input.attrs['type']).lower() == 'submit':
+                    self.submit_var_name = _input.attrs['name']
+                    self.submit_var_value = _input.attrs['name']
+
+                if any(ext in (_input.attrs['name']).lower() for ext in list):
+                    # this condition is used to check that the field is not of type "password". For example : user_password
+                    if (_input.attrs['type']).lower() == 'text':
+                        self.username_parameter_field.append(_input.attrs['name'])
+
+                if (_input.attrs['type']).lower() == 'password':
+                    self.password_parameter.append(_input.attrs['name'])
+
+        if self.password_parameter:
+            return True
+        else:
+            return False
+
+    def check_success_auth(self, content_response: str):
+        with open(path_join(self.CONFIG_DIR, self.PAYLOADS_SUCCESS), 'r', errors='ignore') as success_pattern_file:
+            for success_pattern in success_pattern_file:
+                if success_pattern.strip("\n") in content_response:
+                    return True
+        with open(path_join(self.CONFIG_DIR, self.PAYLOADS_FAIL), 'r', errors='ignore') as fail_pattern_file:
+            for fail_pattern in fail_pattern_file:
+                if fail_pattern in content_response:
+                    return False
```

Comment:
There is no need to look for fail pattern here : if you do not have a success pattern, it is a failure. You can remove these lines since the last return of the function is "return False".